### DISCUSSION TITLE
Queries to find all related entities and relationship for a set of entities.

### DISCUSSION THREAD

**Discussion by xs2tarunkukreja** [vesoft-inc/nebula#5593] [Q&A]
We are using Nebula Graph having N types of entities and M types of relationship.

Scenario - I have a list of entities. Now I want to get all other entities that are linked directly or indirectly to those entities and relationship between them.

For e.g. A -> B - > C -> D->E

Scenario 1 -
Now if I have B.. I want all related entities and relationship like A, C, D, E.

**Comment by wey-gu** [comment]
You could try GET SUBGRAPH on given vertices:

```
GET SUBGRAPHS 2 STEPS FROM "player100","player101" YIELD vertices as entities, edges as relationships;
```

Or with MATCH leveraging the functions: NODES() and RELATIONSHIPS():

```cypher
MATCH p=(x:player)-->()-->()-->()
WHERE id(x) IN ["player100","player101"]
RETURN nodes(p) AS entities, relationships(p) AS relationships
```

**Reply by xs2tarunkukreja** [reply]
Thanks I will try this queries. 

You mentioned 2 steps and -->()-->()-->(). It means we will have subgraph up-to 2 paths only. Am I correct?

**Reply by xs2tarunkukreja** [reply]
A->B->C <- D

If the input is B in above query then will it return A as well?
Will it return D as well?

If not then what change should I make to the query to get A and D?

**Reply by wey-gu** [reply]
Sorry the patterns(`(x:player)-->()-->()-->()` or `2 STEPS FROM "player100","player101"`) are just examples, you should compose as your actual situation.

**Reply by xs2tarunkukreja** [reply]
It means we should know the DB relationship structure... There is no query where we can have subgraph for which particular vertex is part of.

**Reply by wey-gu** [reply]
In that case we could use match, and `return id(v), nodes(p) AS entities, relationships(p) AS relationships` so that it's grouped by your starting node?
